### PR TITLE
docs: update history related

### PR DESCRIPTION
**Pull request by daixiang0**
Signed-off-by: Loong Dai <loong.dai@intel.com>

After #21075, no `version_history/current.rst` any longer.

Commit Message: Update history related docs.
Additional Description:
Risk Level:
Testing:
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Fixes commit #PR or SHA]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

File: RELEASES.md [hunk 1]
```diff
@@ -129,45 +129,37 @@ deadline of 3 weeks.
...
-----------------------
-*Changes that may cause incompatibilities for some users, but should not for most*
+bug_fixes:
+// Changes expected to improve the state of the world and are unlikely to have negative effects*
 
-Bug Fixes
----------
-*Changes expected to improve the state of the world and are unlikely to have negative effects*
+removed_config_or_runtime:
+// Normally occurs at the end of the* :ref:`deprecation period <deprecated>`
```
Comment (phlax) [inline RELEASES.md:152]: can you prefix these comments with `#` - im not sure that `//` is a valid yaml comment, but either way i think better use hashes

**Review by phlax — CHANGES_REQUESTED** [review]
@daixiang0 thanks for this

one small nit in the examples and we should be good to land

File: RELEASES.md (hunk 1 above)
Comment (daixiang0) [reply RELEASES.md]: Sure.

**Review by phlax — APPROVED** [review]
lgtm, thanks @daixiang0

**Pull request merged**